pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
We still need this check, and yes, it's the wrong place to handle it. We should do this in `NewRSNewReplicas` when calculating `scaleUpCount`. Right now we don't prevent this value from being negative which means we could end up with scaled down new ReplicaSet. 

Removing this code doesn't cause any test failure also indicates that we probably need a test for this case.